lldb/source/API/SBWatchpointOptions.cpp

```diff
@@ -0,0 +1,67 @@
+//===-- SBWatchpointOptions.cpp -------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "lldb/API/SBWatchpointOptions.h"
+#include "lldb/Breakpoint/Watchpoint.h"
+#include "lldb/Utility/Instrumentation.h"
+
+#include "Utils.h"
+
+using namespace lldb;
+using namespace lldb_private;
+
+class WatchpointOptionsImpl {
+public:
+  bool m_read = false;
+  bool m_write = false;
+  bool m_modify = false;
+};
```

Comment:
This could be moved to Watchpoint.h and then used in our internal API as well if you like that idea.

Reply:
Yeah I originally wrote that little class in Watchpoint.h and then I moved it into the SBWatchpointOptions because I wasn't using it anywhere else.  tbh I thought the class for the three bools was overkill right now but I agree that there's long-term value in making the SB API more extensible.  But for the lldb_private API, I'm less worried about using three bools for now.  If it does become more than that, an Options class would be a thing to adopt.

Reply:
Yeah, we can do anything internally, so it doesn't really matter what we do inside as we can always change it at any point as long as the public API is solid.